pkg/reconciler/autoscaling/kpa/kpa.go

```diff
@@ -267,13 +268,13 @@ func computeActiveCondition(pa *pav1alpha1.PodAutoscaler, pc podCounts) {
 }
 
 // activeThreshold returns the scale required for the pa to be marked Active
-func activeThreshold(pa *pav1alpha1.PodAutoscaler) int {
+func activeThreshold(ctx context.Context, pa *pav1alpha1.PodAutoscaler) int {
 	min, _ := pa.ScaleBounds()
-	if min < 1 {
-		min = 1
+	if !pa.Status.HasBeenActive() {
+		initialScale := resources.GetInitialScale(config.FromContext(ctx).Autoscaler, pa)
+		return int(math.Max(float64(min), float64(initialScale)))
```

Comment:
I think rather than doing all these casts we can just write `intMax` function, which we might already have.